src/orange/BoundingBoxUtils.hh

```diff
@@ -180,6 +163,24 @@ calc_intersection(BoundingBox<T> const& a, BoundingBox<T> const& b)
     return BoundingBox<T>::from_unchecked(lower, upper);
 }
 
+//---------------------------------------------------------------------------//
+/*!
+ * Check if all points inside the small bbox are in the big bbox.
+ *
+ * All bounding boxes should enclose a "null" bounding box (there are no points
+ * in the null box, so no points are outside the big box). The null bounding
+ * box will enclose nothing but a bounding box.
+ */
+template<class T>
+inline bool encloses(BoundingBox<T> const& big, BoundingBox<T> const& small)
+{
+    auto axes = range(to_int(Axis::size_));
+    return all_of(axes.begin(), axes.end(), [&big, &small](int ax) {
```

Comment:
Just curious why you unrolled for is_inside, but not here.

Reply:
It's a little arbitrary I admit... but:
- `is_inside` is easier to write unrolled
- it's the only one used at runtime (running without optimization means it'll be a bit faster, a dumb reason)
- unrolling means not having to include the algorithms header